solidity/dashboard/src/services/tokens-page.service.js

```diff
@@ -374,11 +365,86 @@ const getGranteeDelegations = async (
     false
   )
 
-  return await getDelegations(
+  const [delegations, undelegations] = await getDelegations(
     allOperators,
     initializationPeriod,
     undelegationPeriod
   )
+
+  return [delegations, undelegations, granteeGrants]
+}
+
+export const getCopiedDelegations = async (
+  ownerOrGrantee,
+  grantIds,
+  initializationPeriod,
+  undelegationPeriod
+) => {
+  const {
+    stakingPortBackerContract,
+    grantContract,
+    stakingContract,
+  } = await ContractsLoaded
+
+  const copiedEvents = await stakingPortBackerContract.getPastEvents(
+    "StakeCopied",
+    {
+      fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.stakingPortBackerContract,
+      filter: { owner: ownerOrGrantee },
+    }
+  )
+
+  const operatorsToCheck = copiedEvents.map(
+    (event) => event.returnValues.operator
+  )
+
+  // We only want operators for whom the delegation has not been paid back.
+  const operatorsOfPortBacker = await getOperatorsOfOwner(
+    stakingPortBackerContract.options.address,
+    operatorsToCheck
```

Comment:
We should probably add a check for empty `operatorsToCheck` similar as we did in other places. I'm having an issue where all accounts are seeing copied delegation.

Reply:
When we add this check I believe we should include in the if statement most of the following code, as it all relates to having nonempty `operatorsToCheck`.

Reply:
yeah.... I've tested locally with `Ganache` and eg. ` await stakingPortBackerContract.getPastEvents("StakeCopied", { fromBlock: 0, filter: { operator: [] }})` returns empty array but `geth` returns all emitted events... so I couldn't reproduce it locally.

I'll update...